charts/k3s/templates/deployment.yaml

```diff
@@ -0,0 +1,348 @@
+{{- if .Values.enableHA }}
```

Comment:
Sure, we can also adjust the StatefulSet accordingly. I initially implemented it as Deployment since we do not need the strong guarantees provided by StatefulSets.

Reply:
I guess in this particular instance we could just switch between statefulset and deployment, as only the `spec.strategy` and `spec.volumeClaimTemplates` and `spec.serviceName` actually change.

Reply:
Okay, got it. Should I add the changes to the template `statefulset.yaml` or replace the template with something more neutral like `workload.yaml` since we switch between Deployment and StatefulSet based on the set values?

Reply:
Good question, I guess `statefulset.yaml` is still okay as its by default a statefulset and we could still change it later if that should cause confusion